lib/AST/ASTImporter.cpp

```diff
@@ -708,6 +711,22 @@ ASTNodeImporter::import(TemplateParameterList *From) {
       *ToRequiresClause);
 }
 
+Error ASTNodeImporter::ImportTemplateParameterLists(const DeclaratorDecl *FromD,
```

Comment:
We skiip the import of the Requires Clause with this refactor, don't we?

Reply:
Also, how is the refactor of this function is related to the source locations?

Reply:
There should be a `import` with `TemplateParameterList` argument that does the same thing as `ImportTemplateParameterList` (as the FIXME: comment says, this function should be replaced with it). Can be done in a separate patch but there was some related other change.

The `ImportTemplateParameterLists` is a new function that imports the `TemplateParameterList`'s from a `Decl` that contains a collection of these. It calls the import of TemplateParameterList (that should be same as `ImportTemplateParameterList`). Missing import of TemplateParameterLists resulted in different SourceLocations if the (start or end) location was taken from the first or last template parameter (that was missing).